packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -327,6 +335,27 @@ export default async function initSanity(
   // Ensure we are using the output path provided by user
   outputPath = answers.outputPath
 
+  if (isNextJs) {
+    const packageJson = readPackageJson(`${outputPath}/package.json`)
+    const reactVersion = packageJson?.dependencies?.react
+
+    if (reactVersion) {
+      const isUsingReact19 = semver.coerce(reactVersion)?.major === 19
+      const isUsingNextJs15 =
+        isNextJs && semver.coerce(detectedFramework?.detectedVersion)?.major === 15
```

Comment:
As this is inside of the `isNextJs` branch, this first check should always be true?

```suggestion
        semver.coerce(detectedFramework?.detectedVersion)?.major === 15
```